### PR TITLE
FIX illegal JSON in attributeFormat=object mode

### LINKED ISSUE #2058: Ilegal JSON rendering in attributeFormat=object mode (NGSIv1) when values are numbers
Raised in http://stackoverflow.com/questions/36737772/invalid-json-when-attributesformat-object-is-used-in-the-tourguide

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #2058 

@crbrox 
@kzangeli

### CODE REVIEW

File: src/lib/ngsi/ContextAttribute.cpp [hunk 1]
```diff
@@ -574,13 +574,14 @@ std::string ContextAttribute::renderAsJsonObject
...
-          effectiveValue      = toString(numberValue);
-          valueIsNumberOrBool = true;
+          effectiveValue = toString(numberValue);
+          withoutQuotes  = true;
         }
         break;
 
       case ValueTypeNone:
         effectiveValue = "null";
+        withoutQuotes  = true;
```
Comment (fgalan) [inline src/lib/ngsi/ContextAttribute.cpp]: I could have used  `valueIsNumberOrBoolOrNull` but that's too much, so I opted for `withoutQuotes`.

**Comment by kzangeli** [comment]
LGTM, but merge with develop needed

File: src/lib/ngsi/ContextAttribute.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/ngsi/ContextAttribute.cpp]: withoutQuotes is ok.
isString (the negation) would be ok as well (I guess).

NTC

File: test/functionalTest/cases/2058_wrong_number_rendering/ilegal_json_attributeformat_object_with_number.test [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Illegal JSON rendering for number attribute values with attributeFormat object
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255
+
+--SHELL--
+
+#
+# 01. Create entity with not string types
```
Comment (kzangeli) [inline test/functionalTest/cases/2058_wrong_number_rendering/ilegal_json_attributeformat_object_with_number.test]: not string types ...
types that are not 'string' ?

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -16,3 +16,5 @@
 - Fix: NGSIv2 subscription payload (#2014)
 - Add: string support in <, >, <=, >= and range filters (#1945)
 - Add: NGSIv2 notifications (Issue #1875)
+- Fix: 'null' attribute value not surrounded by quotes (") in NGSIv1 renderings (#2058)
+- Fix: not-string attribute values were breaking JSON in attributeFormat=object mode (#2058)
```
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: non-string attribute values

Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Fixed in 079c814af67bbc6d192f13df6639e6f440d6c43f

File: test/functionalTest/cases/2058_wrong_number_rendering/ilegal_json_attributeformat_object_with_number.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/2058_wrong_number_rendering/ilegal_json_attributeformat_object_with_number.test]: Fixed in 2acc01f399fffef17b49d6116ecf1273f01be009

**Comment by kzangeli** [comment]
LGTM, again :-)

**Comment by crbrox** [comment]
LGTM

**Pull request merged**